Jakarta: master build throws NoClassDefFoundError at startup

### Describe the bug

Starting Wildfly 27 web app with primefaces-jakarta throws a NoClassDefFoundError for a **javax** class. Seems that there is a wrong dependency,

`Cause: Class 'org.primefaces.component.accordionpanel.AccordionPanelRenderer' is missing a runtime dependency: java.lang.NoClassDefFoundError: javax/el/PropertyNotFoundException

2023-01-13 08:28:49,620 ERROR [stderr] (ServerService Thread Pool -- 109) 	at com.sun.jsf-impl@4.0.0.SP01//com.sun.faces.config.processor.AbstractConfigProcessor.createInstance(AbstractConfigProcessor.java:246)

2023-01-13 08:28:49,620 ERROR [stderr] (ServerService Thread Pool -- 109) 	at com.sun.jsf-impl@4.0.0.SP01//com.sun.faces.config.processor.AbstractConfigProcessor.createInstance(AbstractConfigProcessor.java:191)

2023-01-13 08:28:49,621 ERROR [stderr] (ServerService Thread Pool -- 109) 	at com.sun.jsf-impl@4.0.0.SP01//com.sun.faces.config.processor.RenderKitConfigProcessor.addRenderers(RenderKitConfigProcessor.java:285)

2023-01-13 08:28:49,621 ERROR [stderr] (ServerService Thread Pool -- 109) 	at com.sun.jsf-impl@4.0.0.SP01//com.sun.faces.config.processor.RenderKitConfigProcessor.process(RenderKitConfigProcessor.java:162)

2023-01-13 08:28:49,621 ERROR [stderr] (ServerService Thread Pool -- 109) 	at com.sun.jsf-impl@4.0.0.SP01//com.sun.faces.config.ConfigManager.lambda$initialize$1(ConfigManager.java:295)

2023-01-13 08:28:49,621 ERROR [stderr] (ServerService Thread Pool -- 109) 	at java.base/java.util.AbstractList$RandomAccessSpliterator.forEachRemaining(AbstractList.java:720)

2023-01-13 08:28:49,621 ERROR [stderr] (ServerService Thread Pool -- 109) 	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
...
2023-01-13 08:28:49,623 ERROR [stderr] (ServerService Thread Pool -- 109) Caused by: java.lang.NoClassDefFoundError: javax/el/PropertyNotFoundException

2023-01-13 08:28:49,623 ERROR [stderr] (ServerService Thread Pool -- 109) 	at java.base/java.lang.Class.forName0(Native Method)

2023-01-13 08:28:49,623 ERROR [stderr] (ServerService Thread Pool -- 109) 	at java.base/java.lang.Class.forName(Class.java:467)
...
`

### Reproducer

_No response_

### Expected behavior

_No response_

### PrimeFaces edition

Community

### PrimeFaces version

13.0.0-SNAPSHOT

### Theme

_No response_

### JSF implementation

Mojarra

### JSF version

4.0

### Java version

17

### Browser(s)

_No response_

--- a/primefaces/src/main/java/org/primefaces/component/datatable/feature/SortFeature.java
+++ b/primefaces/src/main/java/org/primefaces/component/datatable/feature/SortFeature.java
@@ -23,23 +23,27 @@
  */
 package org.primefaces.component.datatable.feature;
 
+import java.io.IOException;
+import java.text.Collator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import javax.el.ValueExpression;
+import javax.faces.FacesException;
+import javax.faces.context.FacesContext;
+import javax.faces.model.ListDataModel;
+
 import org.primefaces.PrimeFaces;
 import org.primefaces.component.datatable.DataTable;
 import org.primefaces.component.datatable.DataTableRenderer;
 import org.primefaces.component.datatable.DataTableState;
 import org.primefaces.event.data.PostSortEvent;
-import org.primefaces.model.*;
-
-import javax.faces.FacesException;
-import javax.faces.context.FacesContext;
-import javax.faces.model.ListDataModel;
-import java.io.IOException;
-import java.text.Collator;
-import java.util.*;
-import java.util.concurrent.atomic.AtomicInteger;
-import java.util.stream.Collectors;
-import java.util.stream.IntStream;
-import javax.el.ValueExpression;
+import org.primefaces.model.SortMeta;
+import org.primefaces.model.SortOrder;
 import org.primefaces.util.ComponentUtils;
 
 public class SortFeature implements DataTableFeature {
@@ -279,7 +283,8 @@ public class SortFeature implements DataTableFeature {
             return (List<T>) ((ListDataModel<T>) value).getWrappedData();
         }
         else {
-            throw new FacesException("Data type should be java.util.List or javax.faces.model.ListDataModel instance to be sortable.");
+            throw new FacesException(String.format("Data type should be java.util.List or %s instance to be sortable."
+                    , javax.faces.model.ListDataModel.class.getName()));
         }
     }
 }

--- a/primefaces/src/main/java/org/primefaces/validate/bean/AbstractClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/AbstractClientValidationConstraint.java
@@ -23,14 +23,13 @@
  */
 package org.primefaces.validate.bean;
 
-import javax.validation.metadata.ConstraintDescriptor;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Objects;
 
-public abstract class AbstractClientValidationConstraint implements ClientValidationConstraint {
+import javax.validation.metadata.ConstraintDescriptor;
 
-    public static final String CONSTRAINT_PACKAGE = "javax.validation.constraints";
+public abstract class AbstractClientValidationConstraint implements ClientValidationConstraint {
 
     private final String messageId;
     private final String messageMetadata;

--- a/primefaces/src/main/java/org/primefaces/validate/bean/AssertFalseClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/AssertFalseClientValidationConstraint.java
@@ -27,8 +27,8 @@ public class AssertFalseClientValidationConstraint extends AbstractClientValidat
 
     public static final String CONSTRAINT_ID = "AssertFalse";
     public static final String MESSAGE_METADATA = "data-p-afalse-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public AssertFalseClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

--- a/primefaces/src/main/java/org/primefaces/validate/bean/AssertTrueClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/AssertTrueClientValidationConstraint.java
@@ -27,8 +27,8 @@ public class AssertTrueClientValidationConstraint extends AbstractClientValidati
 
     public static final String CONSTRAINT_ID = "AssertTrue";
     public static final String MESSAGE_METADATA = "data-p-atrue-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public AssertTrueClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

--- a/primefaces/src/main/java/org/primefaces/validate/bean/DecimalMaxClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/DecimalMaxClientValidationConstraint.java
@@ -23,16 +23,16 @@
  */
 package org.primefaces.validate.bean;
 
-import org.primefaces.util.HTML;
-
 import java.util.Map;
+
+import org.primefaces.util.HTML;
 
 public class DecimalMaxClientValidationConstraint extends AbstractClientValidationConstraint {
 
     public static final String CONSTRAINT_ID = "DecimalMax";
     public static final String MESSAGE_METADATA = "data-p-decimalmax-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public DecimalMaxClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

--- a/primefaces/src/main/java/org/primefaces/validate/bean/DecimalMinClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/DecimalMinClientValidationConstraint.java
@@ -23,16 +23,16 @@
  */
 package org.primefaces.validate.bean;
 
-import org.primefaces.util.HTML;
-
 import java.util.Map;
+
+import org.primefaces.util.HTML;
 
 public class DecimalMinClientValidationConstraint extends AbstractClientValidationConstraint {
 
     public static final String CONSTRAINT_ID = "DecimalMin";
     public static final String MESSAGE_METADATA = "data-p-decimalmin-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public DecimalMinClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

--- a/primefaces/src/main/java/org/primefaces/validate/bean/DigitsClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/DigitsClientValidationConstraint.java
@@ -23,16 +23,16 @@
  */
 package org.primefaces.validate.bean;
 
-import org.primefaces.util.HTML;
-
 import java.util.Map;
+
+import org.primefaces.util.HTML;
 
 public class DigitsClientValidationConstraint extends AbstractClientValidationConstraint {
 
     public static final String CONSTRAINT_ID = "Digits";
     public static final String MESSAGE_METADATA = "data-p-digits-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public DigitsClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

--- a/primefaces/src/main/java/org/primefaces/validate/bean/EmailClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/EmailClientValidationConstraint.java
@@ -27,8 +27,8 @@ public class EmailClientValidationConstraint extends AbstractClientValidationCon
 
     public static final String CONSTRAINT_ID = "Email";
     public static final String MESSAGE_METADATA = "data-p-email-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public EmailClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

--- a/primefaces/src/main/java/org/primefaces/validate/bean/FutureClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/FutureClientValidationConstraint.java
@@ -27,8 +27,8 @@ public class FutureClientValidationConstraint extends AbstractClientValidationCo
 
     public static final String CONSTRAINT_ID = "Future";
     public static final String MESSAGE_METADATA = "data-p-future-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public FutureClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

--- a/primefaces/src/main/java/org/primefaces/validate/bean/FutureOrPresentClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/FutureOrPresentClientValidationConstraint.java
@@ -27,8 +27,8 @@ public class FutureOrPresentClientValidationConstraint extends AbstractClientVal
 
     public static final String CONSTRAINT_ID = "FutureOrPresent";
     public static final String MESSAGE_METADATA = "data-p-futureorpresent-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public FutureOrPresentClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

--- a/primefaces/src/main/java/org/primefaces/validate/bean/MaxClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/MaxClientValidationConstraint.java
@@ -23,16 +23,16 @@
  */
 package org.primefaces.validate.bean;
 
-import org.primefaces.util.HTML;
-
 import java.util.Map;
+
+import org.primefaces.util.HTML;
 
 public class MaxClientValidationConstraint extends AbstractClientValidationConstraint {
 
     public static final String CONSTRAINT_ID = "Max";
     public static final String MESSAGE_METADATA = "data-p-max-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public MaxClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

--- a/primefaces/src/main/java/org/primefaces/validate/bean/MinClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/MinClientValidationConstraint.java
@@ -23,16 +23,16 @@
  */
 package org.primefaces.validate.bean;
 
-import org.primefaces.util.HTML;
-
 import java.util.Map;
+
+import org.primefaces.util.HTML;
 
 public class MinClientValidationConstraint extends AbstractClientValidationConstraint {
 
     public static final String CONSTRAINT_ID = "Min";
     public static final String MESSAGE_METADATA = "data-p-min-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public MinClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

--- a/primefaces/src/main/java/org/primefaces/validate/bean/NegativeClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/NegativeClientValidationConstraint.java
@@ -31,8 +31,8 @@ public class NegativeClientValidationConstraint extends AbstractClientValidation
 
     public static final String CONSTRAINT_ID = "Negative";
     public static final String MESSAGE_METADATA = "data-p-negative-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
     public static final String MAX_VALUE = "-0.0000001";
 
     public NegativeClientValidationConstraint() {

--- a/primefaces/src/main/java/org/primefaces/validate/bean/NegativeOrZeroClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/NegativeOrZeroClientValidationConstraint.java
@@ -31,8 +31,8 @@ public class NegativeOrZeroClientValidationConstraint extends AbstractClientVali
 
     public static final String CONSTRAINT_ID = "NegativeOrZero";
     public static final String MESSAGE_METADATA = "data-p-negativeorzero-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
     public static final String MAX_VALUE = "0";
 
     public NegativeOrZeroClientValidationConstraint() {

--- a/primefaces/src/main/java/org/primefaces/validate/bean/NotBlankClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/NotBlankClientValidationConstraint.java
@@ -27,8 +27,8 @@ public class NotBlankClientValidationConstraint extends AbstractClientValidation
 
     public static final String CONSTRAINT_ID = "NotBlank";
     public static final String MESSAGE_METADATA = "data-p-notblank-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public NotBlankClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

--- a/primefaces/src/main/java/org/primefaces/validate/bean/NotEmptyClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/NotEmptyClientValidationConstraint.java
@@ -27,8 +27,8 @@ public class NotEmptyClientValidationConstraint extends AbstractClientValidation
 
     public static final String CONSTRAINT_ID = "NotEmpty";
     public static final String MESSAGE_METADATA = "data-p-notempty-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public NotEmptyClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

--- a/primefaces/src/main/java/org/primefaces/validate/bean/NotNullClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/NotNullClientValidationConstraint.java
@@ -27,8 +27,8 @@ public class NotNullClientValidationConstraint extends AbstractClientValidationC
 
     public static final String CONSTRAINT_ID = "NotNull";
     public static final String MESSAGE_METADATA = "data-p-notnull-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public NotNullClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

--- a/primefaces/src/main/java/org/primefaces/validate/bean/NullClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/NullClientValidationConstraint.java
@@ -27,8 +27,8 @@ public class NullClientValidationConstraint extends AbstractClientValidationCons
 
     public static final String CONSTRAINT_ID = "Null";
     public static final String MESSAGE_METADATA = "data-p-null-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public NullClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

--- a/primefaces/src/main/java/org/primefaces/validate/bean/PastClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/PastClientValidationConstraint.java
@@ -27,8 +27,8 @@ public class PastClientValidationConstraint extends AbstractClientValidationCons
 
     public static final String CONSTRAINT_ID = "Past";
     public static final String MESSAGE_METADATA = "data-p-past-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public PastClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

--- a/primefaces/src/main/java/org/primefaces/validate/bean/PastOrPresentClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/PastOrPresentClientValidationConstraint.java
@@ -27,8 +27,8 @@ public class PastOrPresentClientValidationConstraint extends AbstractClientValid
 
     public static final String CONSTRAINT_ID = "PastOrPresent";
     public static final String MESSAGE_METADATA = "data-p-pastorpresent-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public PastOrPresentClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

--- a/primefaces/src/main/java/org/primefaces/validate/bean/PatternClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/PatternClientValidationConstraint.java
@@ -23,16 +23,16 @@
  */
 package org.primefaces.validate.bean;
 
-import org.primefaces.util.HTML;
-
 import java.util.Map;
+
+import org.primefaces.util.HTML;
 
 public class PatternClientValidationConstraint extends AbstractClientValidationConstraint {
 
     public static final String CONSTRAINT_ID = "Pattern";
     public static final String MESSAGE_METADATA = "data-p-pattern-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public PatternClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

--- a/primefaces/src/main/java/org/primefaces/validate/bean/PositiveClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/PositiveClientValidationConstraint.java
@@ -31,8 +31,8 @@ public class PositiveClientValidationConstraint extends AbstractClientValidation
 
     public static final String CONSTRAINT_ID = "Positive";
     public static final String MESSAGE_METADATA = "data-p-positive-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
     public static final String MIN_VALUE = "0.0000001";
 
     public PositiveClientValidationConstraint() {

--- a/primefaces/src/main/java/org/primefaces/validate/bean/PositiveOrZeroClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/PositiveOrZeroClientValidationConstraint.java
@@ -31,8 +31,8 @@ public class PositiveOrZeroClientValidationConstraint extends AbstractClientVali
 
     public static final String CONSTRAINT_ID = "PositiveOrZero";
     public static final String MESSAGE_METADATA = "data-p-positiveorzero-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
     public static final String MIN_VALUE = "0";
 
     public PositiveOrZeroClientValidationConstraint() {

--- a/primefaces/src/main/java/org/primefaces/validate/bean/SizeClientValidationConstraint.java
+++ b/primefaces/src/main/java/org/primefaces/validate/bean/SizeClientValidationConstraint.java
@@ -23,16 +23,16 @@
  */
 package org.primefaces.validate.bean;
 
-import org.primefaces.util.HTML;
-
 import java.util.Map;
+
+import org.primefaces.util.HTML;
 
 public class SizeClientValidationConstraint extends AbstractClientValidationConstraint {
 
     public static final String CONSTRAINT_ID = "Size";
     public static final String MESSAGE_METADATA = "data-p-size-msg";
-    public static final String CONSTRAINT_CLASS_NAME = CONSTRAINT_PACKAGE  + "." + CONSTRAINT_ID;
-    public static final String MESSAGE_ID = "{" + CONSTRAINT_CLASS_NAME + ".message}";
+    public static final String CONSTRAINT_CLASS_NAME = String.format("%s.%s", javax.validation.constraints.NotNull.class.getPackage().getName(), CONSTRAINT_ID);
+    public static final String MESSAGE_ID =  String.format("{%s.%s}", CONSTRAINT_CLASS_NAME, ATTR_MESSAGE);
 
     public SizeClientValidationConstraint() {
         super(MESSAGE_ID, MESSAGE_METADATA);

